frontend-react/src/utils/misc.ts

```diff
@@ -174,3 +174,31 @@ export const groupBy = <T>(
         (acc[predicate(value, index, array)] ||= []).push(value);
         return acc;
     }, {} as { [key: string]: T[] });
+
+/* Takes a url that contains the 'report/' location and returns
+    the folder location, sending org, and filename
+*/
+export const parseFileLocation = (
+    urlFileLocation: string
+): {
+    folderLocation: string;
+    sendingOrg: string;
+    fileName: string;
+} => {
+    let fileLocationFromStr = "";
+    let sendingOrgFromStr = "";
+    let fileNameFromStr = "";
+
+    const fileReportsLocation = urlFileLocation.split("/").pop();
+    if (fileReportsLocation?.split("%2F").length === 3) {
```

Comment:
Because of the four `split` calls, we're still creating four separate arrays from splitting on the same string.  I think we can simplify this to be the following:

```jsx
const fileReportsLocation = urlFileLocation.split("/").pop() || "";
const [folderLocation, sendingOrg, fileName] = fileReportsLocation.split("%2F");
        
if (!(folderLocation && sendingOrg && fileName)) {
    return {
        folderLocation: "",
        sendingOrg: "",
        fileName: ""
    }
}

return {
    folderLocation,
    sendingOrg,
    fileName
};
```

Reply:
Might as well instantiate an object with those values at the beginning like:

```
const fileDetails = { 
  folderLocation: "", 
  sendingOrg: "", 
  fileName: "" 
}
const fileReportsLocation = urlFileLocation.split("/").pop() || ""; 
const [folderLocation, sendingOrg, fileName] = fileReportsLocation.split("%2F");

return {
  ...fileDetails,
  ...folderLocation && { folderLocation: folderLocation },
  ...sendingOrg && { sendingOrg: sendingOrg },
  ...fileName && { fileName: fileName },
}
```

Since we're cool with sending empty strings back up, we might as well send back _any_ data we have. Not just when all 3 are present. What do y'all think?

Reply:
Or I guess (sorry for getting into the weeds here) even fewer lines:

```
const fileReportsLocation = urlFileLocation.split("/").pop() || ""; 
const [folderLocation, sendingOrg, fileName] = fileReportsLocation.split("%2F");

return {
  folderLocation : folderLocation ? folderLocation : "",
  sendingOrg: sendingOrg ? sendingOrg : "",
  fileName: fileName ? fileName : "",
}
```

Sometimes being too clever, is too much 😄

Reply:
I can't resist, we can make it _even simpler_:

```
const fileReportsLocation = urlFileLocation.split("/").pop() || ""; 
const [folderLocation = "", sendingOrg = "", fileName = ""] = fileReportsLocation.split("%2F");

return {
  folderLocation ,
  sendingOrg,
  fileName,
}
```

Reply:
Ah, good point!  I think it should be an all-or-nothing thing so maybe we should update this function to return null if all three fragments aren't available.  Like it doesn't make sense for us to show the bucket if we can't identify a file name -- shouldn't happen because the url should be formatted properly, but better to be safe since the check here is fragile (as we've discussed previously).  What do y'all think?

Reply:
If we have the data I'd say we should show it BUT are you saying if there are data fragments, then the data might be mislabeled since the structure is fragile?

Reply:
Yeah, exactly.  It might also not make sense to show individual pieces of the following:

<img width="467" alt="Screenshot 2023-02-10 at 15 56 31" src="https://user-images.githubusercontent.com/2421042/218195972-3c6b3852-b562-46a8-bd47-8c1cf8d8362c.png">

Like if we only have the `RECEIVE` part, that doesn't really seem too important or actionable for the end user.

If we want to, we could tighten this function up even more to ensure that we're capturing the right pieces:
- Folder location is in one of (receive|process|send|et cetera)
- Sending Organization takes the form of something like `([a-zA-Z0-9])\.([a-zA-Z0-9])`
- File name is a UUID + extension

But I think splitting for now is fine for now, and we can test it against real data to see if there are other edge cases to handle.  Whatchu think?

Reply:
If so, if we need all three, then maybe:

```
const fileDetails = { 
  folderLocation: "", 
  sendingOrg: "", 
  fileName: "" 
}
const fileReportsLocation = urlFileLocation.split("/").pop() || ""; 
const fileReportsData =  fileReportsLocation.split("%2F");

if (fileReportsData.length === 3) {
  [fileDetails.folderLocation, fileDetails.sendingOrg, fileDetails.fileName] = fileReportsData;
}

return fileDetails;
```